tests/app/delivery/test_send_to_providers.py

```diff
@@ -624,6 +643,7 @@ def test_should_send_sms_to_international_providers(
         international=True,
         reply_to_text=sample_template.service.get_default_sms_sender()
     )
+    send_to_providers.provider_cache.clear()
```

Comment:
Hmm, it's not so much that I hate that we have to include these lines to clear the cache, it's more that if we were to write another test, we would likely forget about this and not include it. 

Option. Put all these tests in a class and then use a method setup (https://docs.pytest.org/en/stable/xunit_setup.html#method-and-function-level-setup-teardown) to ensure the cache is cleared. To be super safe you'd add it as a setup to ALL tests where we delete all caches everywhere before any test but I think that is probably overkill.

Reply:
Added a setup function in 624bd1d